### PR TITLE
Try not to panic when seeing error

### PR DESCRIPTION
**Pull request by wxing1292**
**Change log description**  
(2-3 concise points about the changes in this PR. When committing this PR, the committer is expected to copy the content of this section to the merge description box)
* Return error, instead of panic when creating reader / writer

**Purpose of the change**  
(_e.g._, Fixes #666, Closes #1234)

**What the code does**  
(Detailed description of the code changes)

**How to verify it**  
(Optional: steps to verify that the changes are effective)

### CODE REVIEW

File: src/segment/writer.rs [hunk 1]
```diff
@@ -153,7 +153,7 @@ impl SegmentWriter {
                     Ok((reply, connection)) => RetryResult::Success((reply, connection)),
                     Err(e) => {
                         info!("failed to setup append using rawclient due to {:?}", e);
-                        RetryResult::Retry(e)
+                        RetryResult::Fail(e)
```
Comment (tkaitchuck) [inline src/segment/writer.rs:156]: Why this change?

Comment (wxing1292) [reply src/segment/writer.rs]: reverted

made this change to see whether the recover speed will be improved

**Review by tkaitchuck — COMMENTED** [review]
This has some errors. Please fix them so it can be merged.

**Review by tkaitchuck — APPROVED** [review]

**Review by tkaitchuck — CHANGES_REQUESTED** [review]
It appears this introduces some compile errors because not all of the dependent functions have been updated to deal with the new signatures.
